src/schema.rs

```diff
@@ -33,26 +33,68 @@ pub trait SchemaType {
     fn get_type() -> crate::schema::Type;
 }
 
-/// Contains all the contract schemas for a module
+/// Contains all the contract schemas for a V0 module
 #[derive(Debug, Clone)]
-pub struct Module {
-    pub contracts: BTreeMap<String, Contract>,
+pub struct ModuleV0 {
+    pub contracts: BTreeMap<String, ContractV0>,
 }
 
-/// Describes all the schemas of a smart contract.
+/// Contains all the contract schemas for a V1 module
 #[derive(Debug, Clone)]
-pub struct Contract {
+pub struct ModuleV1 {
+    pub contracts: BTreeMap<String, ContractV1>,
+}
+
+/// Describes all the schemas of a V0 smart contract.
+/// The [Default] instance produces an empty schema.
+#[derive(Debug, Default, Clone)]
+pub struct ContractV0 {
     pub state:   Option<Type>,
     pub init:    Option<Type>,
     pub receive: BTreeMap<String, Type>,
 }
 
-impl Contract {
-    pub fn empty() -> Contract {
-        Contract {
-            state:   None,
-            init:    None,
-            receive: BTreeMap::new(),
+/// Describes all the schemas of a V1 smart contract.
+#[derive(Debug, Default, Clone)]
+/// The [Default] instance produces an empty schema.
+pub struct ContractV1 {
+    pub init:    Option<FunctionSchema>,
+    pub receive: BTreeMap<String, FunctionSchema>,
+}
+
+/// Describes the schema of an init or a receive function for V1 contracts.
+#[derive(Debug, Clone)]
+pub enum FunctionSchema {
```

Comment:
I think this should be named `Function` to be consistent with the rest. We usually use it qualified anyhow, so `schema::Function` is clear on its own.